src/meta/src/rpc/ddl_controller.rs

```diff
@@ -1601,9 +1601,29 @@ impl DdlController {
         // 1. Resolve the upstream fragments, extend the fragment graph to a complete graph that
         // contains all information needed for building the actor graph.
 
+        let (snapshot_backfill_info, cross_db_snapshot_backfill_info) =
+            fragment_graph.collect_snapshot_backfill_info()?;
+
+        // check if log store exists for all cross-db upstreams
+        self.metadata_manager
+            .catalog_controller
+            .validate_cross_db_snapshot_backfill(&cross_db_snapshot_backfill_info)
+            .await?;
+
+        let upstream_table_ids = fragment_graph
+            .dependent_table_ids()
+            .iter()
+            .filter(|id| {
+                !cross_db_snapshot_backfill_info
+                    .upstream_mv_table_id_to_backfill_epoch
+                    .contains_key(id)
+            })
+            .cloned()
+            .collect();
+
         let (upstream_root_fragments, existing_actor_location) = self
```

Comment:
It seems that the `upstream_root_actors` obtained from `upstream_root_fragments` will be used in create mview tracker to track the progress. Is it correct to filter out the cross db backfill actor here?

Reply:
Yes, you're right, we can't simply filter out them to avoid panic. `upstream_root_actors` will be used to track the progress along with `dispatchers`, which does not apply to cross db backfill. The progress info cannot be derived from these fields, so I changed to use an existing function that extract the info from field `fragments` instead. I also removed the field `upstream_root_actors` in the corresponding context.